src/ranking.jl

```diff
@@ -171,3 +171,108 @@ Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; sortkwargs...) =
     _rank(_tiedrank!, x, Float64; sortkwargs...)
+
+"""
+    quantilerank(a::RealVector, score)
+    quantilerank(a::RealVector, score::RealVector)
+
+Return the percentile-position of score (0-1) relative to a.
+
+For example, a quantilerank of 80% means that 80% of the values in `a` are below the given score.
+
+# Arguments
+* `a`: Array of values to which `score` is compared.
+* `score`: Value that is compared to the elements in `a`.
+* `method` (optional, default `:rank`):  method used for calculate the quantilerank. Options available (`:rank`, `:weak`, `:strict`, `:mean`).
+    1. `:rank` - Average percentage ranking of score. In case of multiple matches, average the percentage rankings of all matching scores.
+    2. `:weak` - This method corresponds to the definition of a cumulative distribution function. A quantilerank of 80% means that 80% of values are less than or equal to the provided score.
+    3. `:strict` - Similar to :weak, except that only values that are strictly less than the given score are counted.
+    4. `:mean` - The average of the :weak and :strict scores, often used in testing. See https://en.wikipedia.org/wiki/Percentile_rank
+
+Please, check the examples below to a better understand of the methods. 
+
+# Examples
+Three-quarters of the given values lie below a given score:
+```julia
+julia> quantilerank([1, 2, 3, 4], 3)
+0.625
+```
+
+With multiple matches, note how the scores of the two matches, 0.6 and 0.8 respectively, are averaged:
+```julia
+julia> quantilerank([1, 2, 3, 3, 4], 3)   # default: method=:rank
+0.6
+```
+
+Only 2/5 values are strictly less than 3:
+```julia
+julia> quantilerank([1, 2, 3, 3, 4], 3, method=:weak)
+0.8
+```
+
+But 4/5 values are less than or equal to 3:
+```julia
+julia> quantilerank([1, 2, 3, 3, 4], 3, method=:strict)
+0.4
+```
+
+The average between the weak and the strict scores is:
+```julia
+julia> quantilerank([1, 2, 3, 3, 4], 3, method=:mean)
+0.6
+```
+
+Ref: The "Fundamental Statistics for the Behavioral Sciences" book and the `percentileofscore()` function of Scipy was used as references for the `quantilerank()` function.
+"""
+function quantilerank(a::RealVector{T}, score::T; method::Symbol=:rank) where {T<:Real}
+
+    isnan(score) && return NaN
+
+    n = length(a)
+
+    n == 0 && return 1.0
+
+    if method == :rank
+        nl = sum(a .< score)
```

Comment:
I verified that `sum(<(score), a)` has only one allocation and is super efficient, but I couldn't keep the low allocation in the suggestion `@inbounds for x in a`.  
So I chose to use `count(<(score), a)` because of good performance and low allocation. 
Done at commit 11671f4.

Reply:
Yes `sum(<(score), a)` is fast, but calling it just before `sum(==(score), a)` isn't optimal. `for` loops shouldn't allocate, are you sure you tested this correctly? Make sure you don't benchmark in global scope and that no global variables are used.

Reply:
I did what you suggest in commit 9accf35, which now it's ready for review.